File: src/domain/feedback.ts
export interface MarkPostPublishedRequest {
  postId: string;
  revisionId: string;
  platform: string;
  publishedAt: string;
  publicationUrl: string | null;
}

export interface ManualPublicationView {
  publicationId: string;
  postId: string;
  postTitle: string;
  revisionId: string;
  platform: string;
  publishedAt: string;
  publicationUrl: string | null;
  recordedAt: string;
}

export interface PerformanceMetrics {
  impressions: number;
  reactions: number;
  comments: number;
  reposts: number;
  saves: number;
  profileViews: number;
}

export interface RecordPostPerformanceRequest {
  publicationId: string;
  metrics: PerformanceMetrics;
  notes: string;
}

export interface PerformanceRecordView {
  performanceId: string;
  publicationId: string;
  metrics: PerformanceMetrics;
  notes: string;
  recordedAt: string;
}

export interface FeedbackInsightView {
  kind: string;
  statement: string;
  sampleSize: number;
}

export interface FeedbackSnapshotView {
  publications: ManualPublicationView[];
  latestPerformance: PerformanceRecordView[];
  insights: FeedbackInsightView[];
}

File: src-tauri/src/domain/feedback.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct MarkPostPublishedRequest {
    pub post_id: String,
    pub revision_id: String,
    pub platform: String,
    pub published_at: String,
    pub publication_url: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct ManualPublicationView {
    pub publication_id: String,
    pub post_id: String,
    pub post_title: String,
    pub revision_id: String,
    pub platform: String,
    pub published_at: String,
    pub publication_url: Option<String>,
    pub recorded_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Default)]
#[serde(rename_all = "camelCase")]
pub struct PerformanceMetrics {
    pub impressions: u64,
    pub reactions: u64,
    pub comments: u64,
    pub reposts: u64,
    pub saves: u64,
    pub profile_views: u64,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct RecordPostPerformanceRequest {
    pub publication_id: String,
    pub metrics: PerformanceMetrics,
    pub notes: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct PerformanceRecordView {
    pub performance_id: String,
    pub publication_id: String,
    pub metrics: PerformanceMetrics,
    pub notes: String,
    pub recorded_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct FeedbackInsightView {
    pub kind: String,
    pub statement: String,
    pub sample_size: usize,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct FeedbackSnapshotView {
    pub publications: Vec<ManualPublicationView>,
    pub latest_performance: Vec<PerformanceRecordView>,
    pub insights: Vec<FeedbackInsightView>,
}
